### PR TITLE
chore: vector 0.1.0-alpha.1

### LINKED ISSUE #36: Require hostPath for Agent role
Just flagging this to make sure we address it.

_Originally posted by @jszwedko in https://github.com/timberio/helm-charts/pull/24#discussion_r704444245_

### PR DESCRIPTION
**Pull request by spencergilbert**
Signed-off-by: Spencer Gilbert <spencer.gilbert@datadoghq.com>

Closes #36

### CODE REVIEW

File: charts/vector/README.md [hunk 1]
```diff
@@ -32,14 +32,20 @@ helm install --name <RELEASE_NAME> \
...
+
 TODO
 
 ### Uninstalling the Chart
 
 To uninstall/delete the `<RELEASE_NAME>` deployment:
 
 ```bash
-helm delete <RELEASE_NAME> --purge
+helm delete <RELEASE_NAME>
```
Comment (jszwedko) [inline charts/vector/README.md:48]: Just curious why we dropped the `--purge`.

File: charts/vector/README.md [hunk 2]
```diff
@@ -114,15 +101,45 @@ helm install --name <RELEASE_NAME> \
...
+| secrets.generic | object | `{}` | Each Key/Value will be added to the Secret's data key |
+| securityContext | object | `{}` | Specify securityContext on the Vector container |
 | service.enabled | bool | `true` | If true, create and use a Service resource |
 | serviceAccount.create | bool | `true` | If true, create ServiceAccount |
 | serviceAccount.name | string | `nil` | The name of the ServiceAccount to use. |
-| tolerations | list | `[]` | Allow Vector to schedule on tainted nodes (requires Kubernetes >= 1.6) |
+| tolerations | list | `[]` | Allow Vector to schedule on tainted nodes |
 | updateStrategy | object | `{}` | Customize the updateStrategy used to replace Vector Pods |
+
+### HAProxy values
```
Comment (jszwedko) [inline charts/vector/README.md:118]: So shiney.

File: charts/vector/values.yaml [hunk 3]
```diff
@@ -10,23 +10,46 @@
...
   enabled: false
+  # autoscaling.minReplicas -- Minimum replicas for Vector's HPA
   minReplicas: 1
+  # autoscaling.maxReplicas -- Maximum replicas for Vector's HPA
   maxReplicas: 10
+  # autoscaling.targetCPUUtilizationPercentage -- Target CPU utilization for Vector's HPA
   targetCPUUtilizationPercentage: 80
-  # targetMemoryUtilizationPercentage: 80
+  # autoscaling.targetMemoryUtilizationPercentage -- (int) Target memory utilization for Vector's HPA
+  targetMemoryUtilizationPercentage:
```
Comment (jszwedko) [inline charts/vector/values.yaml:51]: Does this just set the value to `null` by default?

**Review by jszwedko — APPROVED** [review]
Looks good! Just left a couple of questions.

File: charts/vector/README.md (hunk 1 above)
Comment (spencergilbert) [reply charts/vector/README.md:48]: `--purge` is only supported on Helm v2 and isn't an option on v3!

File: charts/vector/values.yaml (hunk 3 above)
Comment (spencergilbert) [reply charts/vector/values.yaml:51]: The template doesn't render unless there's a `value` so by default no utilization target for memory, here: https://github.com/timberio/helm-charts/blob/master/charts/vector/templates/hpa.yaml#L24

Comment (jszwedko) [reply charts/vector/values.yaml:51]: Aha 👍

**Pull request merged**